data/dataloader.py

```diff
@@ -0,0 +1,48 @@
+from type import (
+    PreprocessConfig,
+    DatasetCategories,
+    TrainDataset,
+    TestDataset,
+)
+from typing import List, Callable, Tuple, Optional, Union
+from .transformation import transform_dataset
+from datasets.load import load_dataset
+
+from .merge import merge_datasets
+
+
+class DataLoaderBase:
+    def load(self, category: DatasetCategories) -> Union[TrainDataset, TestDataset]:
+        pass
+
+
+class DataLoader(DataLoaderBase):
```

Comment:
should we stick to the convention where we don't have "base" classes, and we just subclass DataLoader itself?

Reply:
Yes, good with me! Baseclass was needed before because I was doing a different thing in Dataloader and in Dataloadermerge (there was a ``transform_`` function)